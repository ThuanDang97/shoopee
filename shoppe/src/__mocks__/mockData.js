export const PRODUCT_ITEM = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: '-25%',
}

export const PRODUCT_DETAIL = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: '-25%',
  quantity: 1,
  introduction:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
}

export const LIST_PRODUCT = [
  {
    id: 1,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '-25%',
  },
  {
    id: 2,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '',
  },
  {
    id: 3,
    imageUrl: '/images/liraEarring.webp',
    name: 'Kaede Hair Pin Set Of 3',
    price: 20,
    status: '',
  },
  {
    id: 4,
    imageUrl: '/images/liraEarring.webp',
    name: 'Hair Pin Set of 3',
    price: 20,
  },
  {
    id: 5,
    imageUrl: '/images/liraEarring.webp',
    name: 'Plaine Necklace',
    price: 20,
    status: 'Sold out',
  },
  {
    id: 6,
    imageUrl: '/images/liraEarring.webp',
    name: 'Yuki Hair Pin Set of 3',
    price: 20,
    status: '',
  },
]

export const CARTS_MOCK = [
  {
    name: 'Lira Earrings',
    price: 20,
    introduction:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. Nunc sed porta augue. Morbi porta tempor odio, in molestie diam bibendum sed.',
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/Img%2001.png?alt=media&token=7437b1da-7f3b-4ecb-9bbf-e0fca1df9d9b',
    status: '- %21',
    id: 1,
    quantity: 17,
  },
  {
    name: 'Hal Earrings',
    price: 25,
    introduction:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
    description:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. Nunc sed porta augue. Morbi porta tempor odio, in molestie diam bibendum sed.',
    imageUrl:
      'https://firebasestorage.googleapis.com/v0/b/image-blog-155e2.appspot.com/o/Img%2002.png?alt=media&token=598ddd11-aa7b-4e7f-ba58-97cf6d8cdb1a',
    status: '',
    id: 2,
    quantity: 28,
  },
]

export const MOCKED_PRODUCTS = [
  {
    id: 1,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '-25%',
  },
  {
    id: 2,
    imageUrl: '/images/liraEarring.webp',
    name: 'Lira Earrings',
    price: 20,
    status: '',
  },
  {
    id: 3,
    imageUrl: '/images/liraEarring.webp',
    name: 'Kaede Hair Pin Set Of 3',
    price: 20,
    status: '',
  },
  {
    id: 4,
    imageUrl: '/images/liraEarring.webp',
    name: 'Hair Pin Set of 3',
    price: 20,
  },
  {
    id: 5,
    imageUrl: '/images/liraEarring.webp',
    name: 'Plaine Necklace',
    price: 20,
    status: 'Sold out',
  },
  {
    id: 6,
    imageUrl: '/images/liraEarring.webp',
    name: 'Yuki Hair Pin Set of 3',
    price: 20,
    status: '',
  },
]

export const MOCK_PRODUCT = {
  id: 1,
  imageUrl:
    'https://firebasestorage.googleapis.com/v0/b/test-1-50371.appspot.com/o/shoppe-products%2FImg-04.png?alt=media&token=641c3cfe-cab6-4498-8632-eb511434e042',
  name: 'Lira Earrings',
  price: 20,
  status: '-25%',
  quantity: 1,
  introduction:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam placerat, augue a volutpat hendrerit, sapien tortor faucibus augue, a maximus elit ex vitae libero. Sed quis mauris eget arcu facilisis consequat sed eu felis. ',
}
