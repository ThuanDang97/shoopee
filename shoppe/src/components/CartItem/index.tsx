import { memo } from 'react'
import Link from 'next/link'
import { CloseIcon } from '@chakra-ui/icons'
import { Flex, Heading, IconButton, Text } from '@chakra-ui/react'

// Types
import { IProductDetail } from '@self-types/index'

// Utils
import { currencyFormat } from '@utils/index'

interface CartItemProps extends IProductDetail {
  onHandleDeleteCart: (id: number) => void
}

const CartItem = ({
  id,
  name,
  quantity,
  price,
  onHandleDeleteCart,
}: CartItemProps) => {
  return (
    <Flex
      key={`cart-item-${id}`}
      alignItems="center"
      borderRadius="8px"
      backgroundColor="gray"
      justifyContent="space-between"
    >
      <Flex padding="5px 10px" flexDirection="column">
        <Heading as="h3" fontSize="base" fontWeight="medium" color="dark">
          <Link href={`/products/${id}`} prefetch={false}>
            {name}
          </Link>
        </Heading>
        <Text color="dark">Quantity: {quantity}</Text>
        <Text textColor="beaver">{currencyFormat(price)}</Text>
      </Flex>
      <IconButton
        color="secondary"
        fontSize="x-small"
        onClick={() => onHandleDeleteCart(id)}
        type="submit"
        aria-label="delete"
        icon={<CloseIcon />}
      />
    </Flex>
  )
}

export default memo(CartItem)
