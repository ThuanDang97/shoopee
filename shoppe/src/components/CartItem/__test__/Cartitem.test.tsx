import { fireEvent, render, screen } from '@utils/testUtils'

// Components
import CartItem from '@components/CartItem'

// Mocks
import { PRODUCT_DETAIL } from '@mocks/mockData'

describe('CartItem component', () => {
  const props = {
    onHandleDeleteCart: jest.fn(),
    ...PRODUCT_DETAIL,
  }

  it('Renders CartItem correctly', () => {
    const { container } = render(<CartItem {...props} />)

    expect(container).toMatchSnapshot()
  })

  it('Should call handle delete', () => {
    render(<CartItem {...props} />)

    const buttonDelete = screen.getByRole('button', {
      name: /delete/i,
    })

    fireEvent.click(buttonDelete)

    expect(props.onHandleDeleteCart).toBeCalled()
    expect(props.onHandleDeleteCart).toBeCalledTimes(1)
  })
})
