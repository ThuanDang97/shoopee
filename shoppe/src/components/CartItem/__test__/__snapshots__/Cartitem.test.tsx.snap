// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem component Renders CartItem correctly 1`] = `
<div>
  <div
    class="css-kbvgy4"
  >
    <div
      class="css-1c9aly0"
    >
      <h3
        class="chakra-heading css-jf0k73"
      >
        <a
          href="/products/1"
        >
          Lira Earrings
        </a>
      </h3>
      <p
        class="chakra-text css-1qtkha6"
      >
        Quantity: 
        1
      </p>
      <p
        class="chakra-text css-1suja3w"
      >
        $ 20.00
      </p>
    </div>
    <button
      aria-label="delete"
      class="chakra-button css-rzypxw"
      type="submit"
    >
      <svg
        aria-hidden="true"
        class="chakra-icon css-onkibi"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
